### PR TITLE
A user cannot self-modify permissions

### PR DESCRIPTION
**Pull request by marsaud**

### CODE REVIEW

File: src/api/user.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ create.params = {
 // Deletes an existing user.
 async function delete_ ({id}) {
   if (id === this.session.get('user_id')) {
-    throw new InvalidParameters('an user cannot delete itself')
+    throw new InvalidParameters('a user cannot delete itself')
```
Comment (julien-f) [inline src/api/user.js]: Are you sure about this?

It feels better but it goes against standard English rules…

File: src/api/user.js [hunk 2]
```diff
@@ -58,6 +58,9 @@ getAll.permission = 'admin'
 // -------------------------------------------------------------------
 
 export async function set ({id, email, password, permission}) {
+  if (permission && id === this.session.get('user_id')) {
+    throw new InvalidParameters('a user cannot modifiy it\'s own permissions')
```
Comment (julien-f) [inline src/api/user.js]: `modify` but `change` may be better.

Comment (julien-f) [inline src/api/user.js]: `permission`, singular.

File: src/api/user.js (hunk 1 above)
Comment (marsaud) [inline src/api/user.js]: https://www.quora.com/Which-is-the-correct-grammar-usage-a-unique-or-an-unique

Comment (julien-f) [inline src/api/user.js]: It feels natural for me too but it's interesting to see the explanation :)

**Pull request merged**